packages/moonstone/Scrollable/Scrollable.js

```diff
@@ -837,6 +856,31 @@ class ScrollableBase extends Component {
 		}
 	}
 
+	// scroll to top button
+
+	showScrollToTopButton = () => this.setState({shouldShowScrollToTopButton: true});
+
+	hideScrollToTopButton = () => this.setState({shouldShowScrollToTopButton: false});
+
+	toggleScrollToTopButton = ({reachedEdgeInfo}) => {
+		const {bottom, left, right} = reachedEdgeInfo;
+		const {direction, showScrollToTopButton} = this.props;
+		const isRtl = this.uiRef.current.props.rtl;
+		const hasReachedEdge = (
+			bottom && (direction === 'both' || direction === 'vertical') ||
+			!isRtl && (right && (direction === 'both' || direction === 'horizontal')) ||
+			isRtl && (left && (direction === 'both' || direction === 'horizontal'))
+		);
+
+		if (showScrollToTopButton && hasReachedEdge) {
+			this.showScrollToTopButton();
+		} else if (!hasReachedEdge) {
```

Comment:
I suspect we'll want to show this button sooner than when we reach the end as it would be useful much earlier for long scrollers. We might need a prop to configure this either absolutely (e.g. after 200px) or relatively (e.g. after 20%). Or maybe it'll be fixed for moonstone since it's opionated. Yet to be decided. :)